spec/u3d/iniparser_spec.rb

```diff
@@ -82,5 +82,36 @@
         end
       end
     end
+
+    describe '.create_linux_ini', focuss: true do
+      context 'existing ini file' do
+        it 'does not rewrite the file' do
+          path = %r{\/.u3d\/ini_files\/unity-1.2.3f4-linux.ini}
+
+          allow(U3dCore::Helper).to receive(:operating_system) { :linux }
+
+          allow(File).to receive(:file?).with(path) { true }
+          expect(File).to_not receive(:open)
+
+          U3d::INIparser.create_linux_ini('1.2.3f4', 12_345, 'http://example.com/')
+        end
+      end
+
+      context 'non existing ini file' do
+        it 'writes the file' do
+          path = %r{Library\/Application Support\/u3d\/ini_files\/unity-1.2.3f4-linux.ini}
+
+          allow(U3dCore::Helper).to receive(:operating_system) { :mac }
+
+          allow(File).to receive(:file?).with(path) { false }
+          file = double('file')
+          allow(File).to receive(:open).with(path, 'wb') { |&block| block.call(file) }
```

Comment:
`.and_yield file` may be more elegant? It would be more consistent with the rest of the test as well

Reply:
Ah yes. I was looking for the right way to do that